### PR TITLE
Fixed signatures + removed unnecessary code duplication

### PR DESCRIPTION
**Pull request by daguerpedro**
- Updated connected reference on readme
- Removed 'stdafx.h' (Let user configure precompiled headers ) 
- Removed duplicated code at constructor (Now calling directly `init` function).
- Fixed mismatching signatures at `CustomSyntax` & `ReadSerialPort` and `IsConnected`

- Build tested on Visual Studio 2022 and G++ (MinGW-W64 x86_64-ucrt-posix-seh, built by Brecht Sanders) 12.2.0

### CODE REVIEW

**Review by dgrfps — APPROVED** [review]
Seems ok, valid changes.

File: simple-serial-port/simple-serial-port/SimpleSerial.cpp [hunk 1]
```diff
@@ -119,7 +75,7 @@ void SimpleSerial::init(const std::string &com_port, DWORD COM_BAUD_RATE)
     }
 }
 
-void SimpleSerial::CustomSyntax(const std::string &syntax_type)
```
Comment (dmicha16) [inline simple-serial-port/simple-serial-port/SimpleSerial.cpp:122]: Why did you remove the const?

File: simple-serial-port/simple-serial-port/SimpleSerial.cpp [hunk 2]
```diff
@@ -174,7 +130,7 @@ void SimpleSerial::CustomSyntax(const std::string &syntax_type)
     }
 }
 
-std::string SimpleSerial::ReadSerialPort(int reply_wait_time, std::string syntax_type)
+std::string SimpleSerial::ReadSerialPort(int reply_wait_time, const std::string& syntax_type)
```
Comment (dmicha16) [inline simple-serial-port/simple-serial-port/SimpleSerial.cpp:133]: Why did you remove the const? 2x

File: simple-serial-port/simple-serial-port/SimpleSerial.cpp [hunk 3]
```diff
@@ -248,7 +204,7 @@ bool SimpleSerial::CloseSerialPort()
     }
 }
 
-bool SimpleSerial::IsConnected() const
+bool SimpleSerial::IsConnected()
```
Comment (dmicha16) [inline simple-serial-port/simple-serial-port/SimpleSerial.cpp:207]: Why did you remove the const? 3x

**Review by dmicha16 — COMMENTED** [review]
Nice work, thanks for the contrib. Left a couple of comments :slightly_smiling_face:

File: simple-serial-port/simple-serial-port/SimpleSerial.cpp (hunk 1 above)
Comment (daguerpedro) [reply simple-serial-port/simple-serial-port/SimpleSerial.cpp:122]: It could be my mistake, but nor Microsoft compiler nor G++ were able to build SimpleSerial because of mismatching references in the header files and in the source.
Back to the class definition we have: 
` void CustomSyntax(std::string syntax_type);  `
But in the function definition:
`void SimpleSerial::CustomSyntax(const std::string &syntax_type) { ... }`
Causing compilers not to find the functions references.

Comment (daguerpedro) [reply simple-serial-port/simple-serial-port/SimpleSerial.cpp:122]: replacing `const string&` with `string` fixed it

File: simple-serial-port/simple-serial-port/SimpleSerial.cpp (hunk 2 above)
Comment (daguerpedro) [reply simple-serial-port/simple-serial-port/SimpleSerial.cpp:133]: In this case i added the `const &` because SimpleSerial.cpp we had:
`std::string SimpleSerial::ReadSerialPort(int reply_wait_time, std::string syntax_type) { ... }`
But in the definition of the class we had:
`std::string ReadSerialPort(int reply_wait_time, const std::string& syntax_type);`
So the parameters should be matching class definition:
`std::string SimpleSerial::ReadSerialPort(int reply_wait_time, const std::string& syntax_type) { ... }`

Comment (dmicha16) [reply simple-serial-port/simple-serial-port/SimpleSerial.cpp:133]: Sure I get it now. Do you think you could change the header function prototypes to use const instead? It's generally a good practice to use consts in as many places possible when you don't want the local passed args to be mutable. :slightly_smiling_face:

Comment (daguerpedro) [reply simple-serial-port/simple-serial-port/SimpleSerial.cpp:133]: Sure, I'll close this pr and change this.
